src/ibex_bluesky_core/callbacks/fitting/__init__.py

```diff
@@ -127,3 +129,62 @@ def update_fit(self) -> None:
                 self.ydata, weights=None if self.yerr is None else self.weight_data, **kwargs
             )
             self.__stale = False
+
+
+def center_of_mass(x: npt.NDArray[np.float64], y: npt.NDArray[np.float64]) -> float:
+    """Compute our own centre of mass.
+
+    Follow these rules:
+    Background does not skew CoM
+    Order of data does not skew CoM
+    Non constant point spacing does not skew CoM
+    Assumes that the peak is positive
+    """
+    # Offset points for any background
+    # Sort points in terms of x
+    arg_sorted = np.argsort(x)
+    x_sorted = np.take_along_axis(x, arg_sorted, axis=None)
+    y_sorted = np.take_along_axis(y - np.min(y), arg_sorted, axis=None)
+
+    # Each point has its own weight given by its distance to its neighbouring point
+    # Edge cases are calculated as x_1 - x_0 and x_-1 - x_-2
+
+    weight = np.empty(0)
+    x_diff = np.diff(x_sorted)
+
+    weight = np.append(weight, [x_diff[0]])
+
+    for x_diff_i in range(1, x_diff.size):
+        w = [(x_diff[x_diff_i] + x_diff[x_diff_i - 1]) / 2]
+        weight = np.append(weight, w)
+
+    weight = np.append(weight, [x_diff[-1]])
+
+    weight /= np.max(weight)  # Normalise weights in terms of max(weights)
+
+    sum_xyw = np.sum(x_sorted * y_sorted * weight)  # Weighted CoM calculation
+    sum_yw = np.sum(y_sorted * weight)
+    com_x = sum_xyw / sum_yw
+
+    return com_x
+
+
+@make_class_safe(logger=logger)  # pyright: ignore (pyright doesn't understand this decorator)
+class PeakStats(_DefaultPeakStats):
```

Comment:
we should use this in our standard callbacks class - if this ticket is merged before #104 i will add it in, if not i think this PR should